pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,141 @@ $title: Use custom JavaScript in AMP pages
 
-## Scope
+### Handle server data
 
-Any DOM elements the custom JavaScript files wish to interact with must be wrapped inside the `<amp-script>` component tags. This includes other AMP components. The `<amp-script>` component considers `document.body` to be the `<amp-script>` element and not the document's `<body>` element.
+AMP allows you to retrieve server data using `amp-list` and to format it using [`amp-mustache`](../../examples/documentation/amp-mustache.html). In cases where mustache templates are insufficient, `amp-script` can fetch the data, format it, and inject the formatted data into the DOM. If you need to massage server data before sending it to `amp-mustache`, an `amp-script` function can be the data source for `amp-list`. See [the documentation](../../../documentation/components/reference/amp-list.md?format=websites#initialization-from-amp-state) for details and a code sample.
 
-If you were to call `document.body.appendChild(document.createElement('span'))` within the script imported into an `<amp-script>` element in the following document:
+### Introduce new capabilities
+
+You can use `amp-script` to leverage areas of the Web API and DOM API that aren't currently accessible to AMP components, or to use these APIs in ways that AMP components don't support. For example, `amp-script` supports `WebSockets` ([example](https://amp.dev/documentation/examples/components/amp-script/#using-a-websocket-for-live-updates)), `localStorage`, and `Canvas`. It supports a wide variety of browser events, so you can listen for events beyond [those that AMP passes to traditional components](https://amp.dev/documentation/guides-and-tutorials/learn/amp-actions-and-events/). And since `amp-script` provides access to the `navigator` object, you can retrieve information about [the user's browser](https://amp.dev/documentation/examples/components/amp-script/#detecting-the-operating-system) or [preferred language](https://amp.dev/documentation/examples/components/amp-script/#personalization).
+
+## amp-script - or amp-bind and amp-list?
+
+For a new AMP developer who's comfortable with JavaScript, it may seem easier to always use `amp-script` than to learn to use `amp-bind` and `amp-list`. But the complete AMP developer is likely to want to know and use all of these components.
+
+`amp-bind` is generally more straightforward for basic interactions, where its tight integration into HTML tags is compelling. In this example, pressing a button changes a text fragment. AMP's data binding makes this straightforward and easy to read:
+
+```html
```

Comment:
Can we make this an executable sample?

Reply:
Oo, I didn't know you could do that! But, hunting around for a bit, I see [you can](https://amp.dev/documentation/guides-and-tutorials/develop/animations/triggering_css_animations).